### PR TITLE
Importing Vignette Library

### PR DESCRIPTION
**Pull request by nmonterroso**
@drsnyder @pchojnacki @sqreek @nandy-andy @aniaMu @alistra  
https://wikia-inc.atlassian.net/browse/SERVICES-114

This PR does a few things:
1. Remove resources, classes, tests, etc. that have to do with serving images
2. Import/update Vignette library from [java-commons](https://github.com/Wikia/java-commons). Allow this library to be deployed to our Archiva server. We probably want to delete this repo when this PR is merged, since Vignette is the only library there.
3. Piggybacking small change that allows consul config to be null, for the case where the env var is missing in `ConsulVariableLookup`

### CODE REVIEW

File: lib/vignette/src/main/java/com/wikia/vignette/UrlConfig.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    bucket = builder.bucket;
+    baseURL = builder.baseURL;
+    domainShardCount = builder.domainShardCount;
+    queryParams = builder.queryParams;
+    imageType = builder.imageType;
+  }
+
+  public static class Builder {
+    private ArrayList<NameValuePair> queryParams = new ArrayList<>();
+    private Boolean isArchive = false;
```
Comment (aniaMu) [inline lib/vignette/src/main/java/com/wikia/vignette/UrlConfig.java]: can isArchive be null? if not then you can use ```boolean```

**Comment by drsnyder** [comment]
lgtm

**Comment by drsnyder** [comment]
I created https://wikia-inc.atlassian.net/browse/PLATFORM-977 to make sure that we prevent any breaking changes from being released.

File: lib/vignette/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+```
+
+## Usage
+
+To use the library, first, create a configuration object. The configuration object is used to locate
+the image.
+
+```java
+UrlConfig config new UrlConfig.Builder()
```
Comment (nandy-andy) [inline lib/vignette/README.md]: Did you miss `=` here, sir? :hushed:

Comment (nmonterroso) [inline lib/vignette/README.md]: Oops! We need unit tests for `README`s :P

Comment (nandy-andy) [inline lib/vignette/README.md]: Hack-a-thon project? :ghost:

**Comment by nandy-andy** [comment]
Late but always: LGTM :+1:

**Pull request merged**